lib/active_storage/service/gcs_service.rb

```diff
@@ -68,6 +68,10 @@ def url_for_direct_upload(key, expires_in:, content_type:, content_length:, chec
     end
   end
 
+  def headers_for_direct_upload(key, content_type:, checksum:, **)
+    { "Content-Type": content_type, "Content-MD5" => checksum }
```

Comment:
Fixed in https://github.com/rails/activestorage/pull/83/commits/042bf7d3043ab5334b4677c255e0bf1f313bff4d